packages/story-editor/src/utils/useCORSProxy.js

```diff
@@ -30,10 +30,38 @@ function useCORSProxy() {
     actions: { getProxyUrl },
   } = useAPI();
   const enableCORSProxy = useFeature('enableCORSProxy');
+
+  /**
+   * Check if the resource can be accessed directly.
+   *
+   * Makes a HEAD request, which in turn triggers a CORS preflight request
+   * in the browser.
+   *
+   * If the request passes, we don't need to do anything.
+   * If it doesn't, it means we need to run the resource through our CORS proxy at all times.
+   *
+   * @type {function(): boolean}
+   */
+  async function checkResourceAccess(link) {
+    let shouldProxy = false;
+    if (!link) {
+      return shouldProxy;
+    }
+    try {
+      await fetch(link, {
+        method: 'HEAD',
+      });
+    } catch (err) {
+      shouldProxy = true;
+    }
+
+    return shouldProxy;
+  }
+
   const getProxiedUrl = useCallback(
     (resource, src) => {
       const { needsProxy } = resource;
-      if (enableCORSProxy && needsProxy) {
+      if (enableCORSProxy && needsProxy && src) {
```

Comment:
I don't particularly like this. If `src` is falsey/null, `getProxiedUrl` should probably not be called on it in the first place

Reply:
I know, but there is nothing wrong with a little bit of harden here. This came up with a issue in unit tests.

Reply:
We should update the tests to use proper data

Reply:
So this function should only work if propper data is passed to it? We should not handle invalid data?

Reply:
That should be handled at a different level

Reply:
If you could point to the unit test you mentioned, that would be helpful.

Reply:
> That should be handled at a different level

That isn't DRY at all? And may result in someone forget to add this check upstream.